frontend/src/metabase/hooks/use-palette.ts

```diff
@@ -0,0 +1,25 @@
+import { useMemo } from "react";
+
+import type { ColorPalette } from "metabase/lib/colors/types";
+
+import { useMantineTheme, type MantineColor } from "../ui";
+
+/**
+ * Gets a color palette from the current Mantine theme.
+ *
+ * Allows palettes to be overridden by the user, primarily via the React embedding SDK.
+ */
+export function usePalette(): ColorPalette {
+  const theme = useMantineTheme();
+
+  return useMemo(() => {
+    const color = (name: MantineColor) => theme.colors[name]?.[0];
```

Comment:
The idea looks good, but Mantine has a function for this.

```suggestion
    const color = theme.fn.themeColor
```

Or you could just inline that. e.g.
```ts
theme.fn.themeColor("white")
```

Reply:
Didn't knew this, thanks!